### PR TITLE
humidifier card: add on-off button

### PR DESCRIPTION
**Pull request by Shulyaka**
## Proposed change

Add on/off toggle to the humidifier card
***UPDATE:*** Toggle replaced with the button, see comments. Images are outdated.

humidifier_with_mode
humidifier_without_mode
humidifier_unavailable

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
demo:
```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by zsarnett** [comment]
We should just make clicking the card toggle the entity instead of adding a toggle element

**Comment by zsarnett** [comment]
We should align the functionality to the light card. Pressing in the middle will toggle

**Comment by Shulyaka** [comment]
Ok, will try to do that

**Comment by Misiu** [comment]
Take a look at the thermostat card:
image
There was a plan to redesign it, but I think that for now, you can mimic the design by adding functional buttons on the bottom.

**Comment by Shulyaka** [comment]
Hi @Misiu 

This was my original idea. But then I realized that since we only have two options (on-off) it would be visually better to replace them with a switch.

Now I am trying to do something like what @zsarnett suggested:

humidifier_wip

It's not the final design though and since I am not a web developer, I am not fast at these things.

**Comment by Shulyaka** [comment]
@zsarnett Please check now

No mouse-over:
humidifier-no-hover

Mouse-over:
humidifier-hover

Mouse-press:
humidifier-click

**Comment by Shulyaka** [comment]
Fixes #11327

**Comment by Shulyaka** [comment]
@zsarnett Any chance it could make it to 2022.10? :)

**Comment by bramkragten** [comment]
@matthiasdebaat could you take a look at this?

**Comment by matthiasdebaat** [comment]
Can you center align the circle to the value? (Left current, right is suggestion)
CleanShot 2022-09-19 at 11 40 54@2x


> Mouse-over: 
> humidifier-hover

The hover state on dark mode is hardly visible, it is consistent with the light card though. Can you make it a bit more visible on both?

**Comment by Shulyaka** [comment]
Sure I can do that, but let me first explain my design decisions.

I wanted the circle to be centered inside the round slider, even moved the number slightly down to fit it inside, and the hover state I chose to be consistent with the light card:
light_card

**Comment by matthiasdebaat** [comment]
Yeah, both ways doesn't really fit: circle isn't in the center of the slider or the percentage isn't centred. 

In this case I would suggest to focus on the percentage, because the circle surrounds it and got the most attention value compared to the slider.

**Comment by Shulyaka** [comment]
@matthiasdebaat Done all your suggestions, and also made the round slider grey when the humidifier is off. Please check.

**Comment by Shulyaka** [comment]
Hover:
image
Pressed:
image
Off:
image
Light theme:
image

**Comment by Shulyaka** [comment]
Once this is merged, I have a couple of other cool features for the humidifier card coming up...

**Comment by matthiasdebaat** [comment]
> @matthiasdebaat Done all your suggestions, and also made the round slider grey when the humidifier is off. Please check.

Looks good, thanks! (sorry was away the last couple of days)

File: src/panels/lovelace/cards/hui-humidifier-card.ts [hunk 1]
```diff
@@ -132,27 +133,28 @@ export class HuiHumidifierCard extends LitElement implements LovelaceCard {
...
-                    ${this.hass!.localize(
-                      `state_attributes.humidifier.mode.${stateObj.attributes.mode}`
-                    ) || stateObj.attributes.mode}
-                  `
-                : ""
-            }
-          </text>
-        </g>
+    `;
+    const currentMode = svg`
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-humidifier-card.ts:137]:
```suggestion
    const currentMode = html`
```

**Review by bramkragten — DISMISSED** [review]

**Comment by Shulyaka** [comment]
Could you re-trigger CI?

**Review by bramkragten — APPROVED** [review]

**Pull request merged**